test/tests/outputs/checkpoint/tests

```diff
@@ -155,4 +155,25 @@
       detail = "be capable of restarting a simulation from the output data."
     []
   []
+  [autosave_timer]
+    type = RunApp
+    expect_out = "1 seconds have elapsed, autosaving checkpoint..."
+    input = autosavetimer.i
+    recover = false
+    max_parallel = 1
+    max_threads = 1
+    requirement = "The system shall output an automatic checkpoint after a specified interval of time."
+    issues = "#12722"
+  []
+  [autosave_timer_system]
+    type = RunApp
+    expect_out = "1 seconds have elapsed, autosaving checkpoint..."
+    input = autosavetimer.i
+    cli_args = "Outputs/cp/test_system=true"
+    recover = false
```

Comment:
I would prefer to even see this working with recover. That might mean you need to make the test longer since it'll be cut in half by the recover testing. Try it out and see.

Reply:
Is recover recovering from the checkpoint objects created in the test spec, or is this that thing that runs it for half the time on its own?

If its the second, I will admit I don't understand what the benefit of the recovery testing is

Reply:
`recover` in the test spec simply means that you don't want the TestHarness to manually split *this* test into two halves. It does make sense to skip it if you've explicitly setup a recover test. However, that's not what you've done here. A manual recover test typically has to not solve the whole problem on the first half (either by overriding the number of steps on the command line, or by passing the special `--half-transient` flag. Then the second half would pass `--recover` on the command line to read the checkpoint and make sure the test continues. 

Now that I think about this a bit more, we probably don't want the TestHarness to help us with the split here as we might not pick up the right checkpoint file (if we are trying to verify that we can recover from the one written out by autosave). With your testing, you are indeed checking that the screen output is working, which is good. What we aren't testing though is if the checkpoint file actually works.